numpyro/handlers.py

```diff
@@ -325,8 +325,8 @@ def __init__(self, fn, rng):
 
     def process_message(self, msg):
         if msg['type'] == 'sample':
-            msg['kwargs']['random_state'] = self.rng
-            self.rng, = random.split(self.rng, 1)
+            self.rng, rng_sample = random.split(self.rng)
+            msg['kwargs']['random_state'] = rng_sample
```

Comment:
the issue here is when `msg['fn']` uses `self.rng` key, it might split the key (e.g. in StudentT distribution)